packages/runtime-core/src/apiLifecycle.ts

```diff
@@ -48,57 +48,29 @@ function injectHook(
   }
 }
 
-export function onBeforeMount(
+const createLifecycleInjector = (lifecycle: LifecycleHooks): Function => function(
   hook: Function,
   target: ComponentInternalInstance | null = currentInstance
 ) {
-  injectHook(LifecycleHooks.BEFORE_MOUNT, hook, target)
+  injectHook(lifecycle, hook, target)
 }
 
-export function onMounted(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.MOUNTED, hook, target)
-}
-
-export function onBeforeUpdate(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.BEFORE_UPDATE, hook, target)
-}
-
-export function onUpdated(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.UPDATED, hook, target)
-}
-
-export function onBeforeUnmount(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.BEFORE_UNMOUNT, hook, target)
-}
-
-export function onUnmounted(
-  hook: Function,
-  target: ComponentInternalInstance | null = currentInstance
-) {
-  injectHook(LifecycleHooks.UNMOUNTED, hook, target)
-}
+export const onBeforeMount = createLifecycleInjector(LifecycleHooks.BEFORE_MOUNT)
+export const onMounted = createLifecycleInjector(LifecycleHooks.MOUNTED)
+export const onBeforeUpdate = createLifecycleInjector(LifecycleHooks.BEFORE_UPDATE)
+export const onUpdated = createLifecycleInjector(LifecycleHooks.UPDATED)
+export const onBeforeUnmount = createLifecycleInjector(LifecycleHooks.BEFORE_UNMOUNT)
+export const onUnmounted = createLifecycleInjector(LifecycleHooks.UNMOUNTED)
 
 export function onRenderTriggered(
-  hook: Function,
```

Comment:
Maybe you can parametrize `createLifecycleInjector<T extends Function>(hook: T,...)` and then overwrite `T` for `onRenderTracked/Triggered`? Would be even more dry

Reply:
I just realized that, but not familiarize with ts, was searching how to do that, thanks

Reply:
On mobile, so can't check it, but probably it would be something like:
```
createLifecycleInjector<T extends Function>(hook: T,...) {...} // maybe = instead of extends

type DebugHandlerOrWhatever = (e: DebuggerEvent): void;

export const onRenderTriggered = createLifecycleInjector<DebugHandler>(LifecycleHooks.RENDER_TRIGGERED)
```

Reply:
@jacekkarczmarczyk Not sure I'm doing it right, could you please help me review https://github.com/vuejs/vue-next/pull/246/commits/2f5dfea989abc9e15badba489a8242ad076a727e

Reply:
As i mentioned I'm not sure if there should be "extends" or "=", but otherwise lgtm

Reply:
Just saw comment above, test fails, I'll check, thanks a lot